src/app/OperationalSessionSetup.cpp

```diff
@@ -292,12 +294,11 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
 }
 
-CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessageProtocolConfig & config)
+CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ResolveResult & result)
 {
+    auto & config = result.mrpRemoteConfig;
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
-    // TODO: Combine LargePayload flag with DNS-SD advertisements from peer.
-    // Issue #32348.
-    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload)
+    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload && result.supportsTcpServer)
```

Comment:
If `result.supportsTcpServer` is false but `mTransportPayloadCapability == TransportPayloadCapability::kLargePayload`, shouldn't this error out?

Reply:
done